### PR TITLE
feat: added Docker container

### PR DESCRIPTION
**Pull request by SteBaum**
#### Which issue(s) this PR fixes:

FixesNone

#### Additional comments:

To make clear that you license your contribution under
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0) and that you give permission to [TOSIT](https://www.tosit.io/),
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

**Comment by SteBaum** [comment]
The solution to deploy VMs through a container is abandoned.

File: Docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  https://rubygems.org/downloads/diffy-3.4.3.gem \
+  https://rubygems.org/downloads/mini_portile2-2.8.8.gem && \
+  gosu tdp vagrant plugin install vagrant-libvirt --plugin-version=0.12.2 && \
+  rm *.gem
+
+# Set tdp as working directory
+WORKDIR /home/tdp/tdp-dev
+
+# Install Ansible-Core 
+RUN pip install ansible-core==2.15.1 --break-system-packages
```
Comment (rpignolet) [inline Docker/Dockerfile]: Add `--no-cache` in order to reduce image size. We can not mount volume during build so the cache is in the result image.

File: README.md [hunk 2]
```diff
@@ -58,3 +59,34 @@ This network use an interface on the host `vboxnet<N>` which need to be delete m
...
+## Containerized launch
+
+The Docker container has a Libvirt client installed which launches the VMs on your host. The only requirement is to have Docker and Libvirt.
+
+- Check the Libvirt GID for your user the command `id`.
+
+- Build the image:
+
+```sh
+docker build -t libvirt-client Docker
```
Comment (rpignolet) [inline README.md]: Change the name with `tdp-vagrant`. I think this is better.
```suggestion
docker build -t tdp-vagrant Docker
```

File: README.md [hunk 3]
```diff
@@ -58,3 +59,34 @@ This network use an interface on the host `vboxnet<N>` which need to be delete m
...
+
+- Run and enter the container with a bind mount to the current host directory :
+
+```sh
+docker run -it --name libvirt --rm \
+-v $PWD:/home/tdp/tdp-dev \
+-v /var/run/libvirt/libvirt-sock:/var/run/libvirt/libvirt-sock \
+-v vagrant-boxes:/home/tdp/.vagrant.d/boxes \
+--env CONTAINER_UID=$(id -u) --env CONTAINER_GID=<libvirt GID> \
+libvirt-client bash
```
Comment (rpignolet) [inline README.md]: Use `tdp-vagrant` for names and get the GID of `libvirt` group via a shell command.
```suggestion
docker run -it --name tdp-vagrant --rm \
-v $PWD:/home/tdp/tdp-dev \
-v /var/run/libvirt/libvirt-sock:/var/run/libvirt/libvirt-sock \
-v vagrant-boxes:/home/tdp/.vagrant.d/boxes \
--env CONTAINER_UID=$(id -u) --env CONTAINER_GID=$(getent group libvirt | cut -d: -f3) \
tdp-vagrant bash
```

File: Docker/Dockerfile (hunk 1 above)
Comment (SteBaum) [reply Docker/Dockerfile]: done

File: README.md (hunk 2 above)
Comment (SteBaum) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (SteBaum) [reply README.md]: done

**Review by rpignolet — APPROVED** [review]

**Pull request merged**